### PR TITLE
[CDAP-14281] Bootstrap Upgrade on React Side

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-14281
Build: https://builds.cask.co/browse/CDAP-UDUT143

Summary of changes:
- Removal of anything that has `-xs-` (ie. float-xs-*, col-xs-*, text-xs-*). Basically for xs, it should be the default. Other sizes still exist.
- Added an explicit html base font size of 13px. The default for the fonts in Bootstrap 4 seemed to be bigger.
- Most of the inconsistency revolve around flex box. Especially in column layout, the row becomes a flex box, so that is something to watch out for in the shared components between React and Angular

### CODE REVIEW

File: cdap-ui/app/cdap/styles/variables.scss [hunk 1]
```diff
@@ -14,60 +14,60 @@
...
 $brand-danger:          rgb(209, 86, 104);  // #d15668
 
+$primary: $brand-primary;
+$success: $brand-success;
+$info: $brand-info;
+$warning: $brand-warning;
+$danger: $brand-danger;
+$secondary: white;
+
+$theme-colors: map-merge(
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/styles/variables.scss:45]: Curious. Is this is being used in this PR or is it something that we anticipate to use in the future?

File: cdap-ui/app/cdap/styles/lib-styles.scss [hunk 2]
```diff
@@ -45,23 +37,19 @@
 @import "~bootstrap/scss/card";
 @import "~bootstrap/scss/breadcrumb";
 @import "~bootstrap/scss/pagination";
-@import "~bootstrap/scss/tags";
+@import "~bootstrap/scss/badge";
 @import "~bootstrap/scss/jumbotron";
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/styles/lib-styles.scss]: Do we still need all these bootstrap styles? Stylings like modal, tables, input group are being used for sure but styling like breadcrumbs, jumbotron etc., are still being used? Could we exclude them to shave off some bytes shipped to the browser?

**Review by ajainarayanan — COMMENTED** [review]
@elfenheart The initial set of changes looks awesome! Have a couple of minor comments.

One ask from me is, could you summarize the changes done in the PR description? Doesn't have to be detailed, just what changed from bootstrap 3-4 and what changes are in this PR and what is still pending. I understand you have organized your commits well so that it conveys the idea but would be easier to read as a short description.

File: cdap-ui/app/cdap/styles/variables.scss (hunk 1 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/styles/variables.scss:45]: @ajainarayanan this is overriding the bootstrap theme colors, specifically the `secondary` color

File: cdap-ui/app/cdap/styles/lib-styles.scss (hunk 2 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/styles/lib-styles.scss]: I'll see what I can do. Shouldn't be a problem. I would probably just comment them out just in case in future we want them.

File: cdap-ui/app/cdap/components/CollapsibleSidebar/index.js [hunk 3]
```diff
@@ -83,7 +83,7 @@ export default class CollapsibleSidebar extends Component {
         <div className={classnames('collapsible-content', { 'show-content': this.state.expanded })}>
           <div
             className={classnames(
-              'collapsible-toggle-tab text-xs-center text-center',
+              'collapsible-toggle-tab text-center text-center',
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/CollapsibleSidebar/index.js]: nit: duplicate classes

File: cdap-ui/app/cdap/components/PipelineTriggers/EnabledTriggersTab/EnabledTriggerRow.js [hunk 4]
```diff
@@ -45,7 +45,7 @@ export default class EnabledTriggerRow extends Component {
 
   renderLoading() {
     return (
-      <div className="text-xs-center text-center">
+      <div className="text-center text-center">
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/PipelineTriggers/EnabledTriggersTab/EnabledTriggerRow.js]: same here

File: cdap-ui/app/cdap/components/TriggeredPipelines/TriggeredPipelineRow.js [hunk 5]
```diff
@@ -44,7 +44,7 @@ export default function TriggeredPipelineRow({
   }
 
   const loadingIcon = (
-    <div className="text-xs-center text-center">
+    <div className="text-center text-center">
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/TriggeredPipelines/TriggeredPipelineRow.js]: same here

File: cdap-ui/app/cdap/components/Header/Header.scss [hunk 6]
```diff
@@ -159,6 +159,11 @@
   }
   .navbar-list-section,
   .navbar-right-section {
+    // button.btn:focus {
+    //   outline: 0;
+    //   box-shadow: 0;
+    // }
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/Header/Header.scss]: Do you need these?

**Comment by tonybach** [comment]
Seems like in our modals the buttons are now right-aligned instead of left-aligned like before. Can you investigate?

Before:
image

After:
image

**Review by tonybach — DISMISSED** [review]
LGTM 👍

**Review by tonybach — APPROVED** [review]
LGTM 👍
